cli/cmd/plugin/unmanaged-cluster/kubeconfig/kubeconfig.go

```diff
@@ -18,6 +18,22 @@ import (
 	clientcmdapilatest "k8s.io/client-go/tools/clientcmd/api/latest"
 )
 
+// GetKubeconfigContext returns the current context for a passed in kubeconfig file
+func GetKubeconfigContext(filepath string) (string, error) {
+	// Build a new client config and get the ConfigAccess
+	config := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		&clientcmd.ClientConfigLoadingRules{ExplicitPath: filepath},
+		&clientcmd.ConfigOverrides{},
+	).ConfigAccess()
+
+	startingConfig, err := config.GetStartingConfig()
+	if err != nil {
+		return "", nil
+	}
+
+	return startingConfig.CurrentContext, nil
```

Comment:
If `CurrentContext` is not set, this returns an empty string with a nil error. So we continue on and later fail stating that the cluster name must be provided. I think we need a check in `cli/cmd/plugin/unmanaged-cluster/config/config.go` for both `err != nil` and `context != ""`.

Reply:
Or maybe better, a check here for something like `if startingConfig.CurrentContext == "" { return "", fmt.Errorf("no current set context") }`

Reply:
^ That's what I was thinking. It'd be alittle weird to pass a kubeconfig with no context set (either a merged `~/.kube/config` or one given by the provider). So erroring out makes sense to me